File: base-saas-service/service/manage-service/src/main/java/com/base/saas/manage/service/system/SysMenuService.java
package com.base.saas.manage.service.system;/**
 * Created by win7 on 2018/6/11.
 */

import com.base.saas.manage.model.ReturnMap;
import com.base.saas.manage.model.system.SysMenu;

import java.util.List;
import java.util.Map;

/**
 * Title :
 * Description : 系统菜单service，隶属于系统模块
 */
public interface SysMenuService {

    /***
     * 获取当前模块下的所有菜单，如果存在parentId表示获取当前菜单下的子菜单
     *
     * @return
     */
    List<SysMenu> getAllMenuList(String moduleId, String parentId) throws Exception;

    //通过菜单id获取菜单信息
    SysMenu getMenuByMenuId(String MenuId) throws Exception;

    //获取模块下所有的菜单信息,如果存在parentId表示获取当前菜单下的子菜单
    List<SysMenu> getAllMenuTree(String moduleId, String parentId) throws Exception;

    //添加系统菜单
    ReturnMap addMenu(SysMenu moduleDetail) throws Exception;

    //修改系统菜单信息
    ReturnMap editMenu(SysMenu moduleDetail) throws Exception;

    //修改系统菜单状态
    ReturnMap updateMenuStatus(String menuId,int status) throws Exception;
}

File: base-saas-system-manager/src/server/services/system-manage-services/sys-menu.service.ts
/*
*manage 系统模块菜单管理 service
*/
import {managerService} from '~/server/controller'
import {NetService} from '~/utils/net.service'
import {Inject, Debounce} from "~/core/decorator";
import {Service} from "~/server/service"

export class SysMenuService extends Service {

  @Inject(NetService)
  private netService: NetService

  /**
   * 查询已启用的模块列表
   */
  @Debounce()
  getAllMenuTree(moduleId, parentId) {
    return this.netService.send({
      server: managerService.moduleDetailController.getAllMenuTree,
      data: {moduleId: moduleId, parentId: parentId},
      loading: true
    })
  }

  @Debounce()
  getChildrenMenuByParentId(userId, parentId) {
    return this.netService.send({
      server: managerService.moduleDetailController.getChildrenMenuByParentId,
      data: {userId: userId, parentId: parentId},
      loading: true
    })
  }

  //表格数据
  getAllMenuDetailList(data) {
    return this.netService.send({
      server: managerService.moduleDetailController.getAllMenuDetailList,
      data: data,
      loading: true
    })
  }

  //添加菜单
  addMenu(data) {
    return this.netService.send({
      server: managerService.moduleDetailController.addMenu,
      data: data,
      loading: true
    })
  }

  //更改状态
  updateMenuStatus(menuId,status) {
    return this.netService.send({
      server: managerService.moduleDetailController.updateMenuStatus,
      data: {menuId,status},
      loading: true
    })
  }

//编辑获取信息
  getMenuById(menuId) {
    return this.netService.send({
      server: managerService.moduleDetailController.getMenuById,
      data: {menuId},
      loading: true
    })
  }

//编辑保存信息
  editMenu(data) {
    return this.netService.send({
      server: managerService.moduleDetailController.editMenu,
      data: data,
      loading: true
    })
  }

}


